### PR TITLE
Add followLinks and fix linkname symlink

### PR DESCRIPTION
**Pull request by LuKks**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -110,15 +111,20 @@ module.exports = class Localdrive {
   }
 
   async symlink (key, linkname) {
+    // { key: '/README.shortcut', linkname: '/README.md' }
```
Comment (LuKks) [inline index.js:114]: Those comments are temporal

**Comment by LuKks** [comment]
@mafintosh Ready!

File: index.js [hunk 2]
```diff
@@ -113,12 +114,15 @@ module.exports = class Localdrive {
...
+    // note: drive has reversed args compared with fs.symlink
     const { filename: pointer } = keyResolve(this.root, key)
-    const { filename: target } = keyResolve(this.root, linkname)
+    const { filename } = keyResolve(this.root, linkname)
 
     const release = await this._lock()
     try {
       await fsp.mkdir(path.dirname(pointer), { recursive: true })
+
+      const target = path.isAbsolute(linkname) ? filename : path.relative(this.root, filename)
```
Comment (mafintosh) [inline index.js:125]: const target = linkname.startsWith('/') ? path.join(this.root, linkname) : linkname.replace(/\//g, path.sep)

File: index.js [hunk 3]
```diff
@@ -28,8 +29,8 @@ module.exports = class Localdrive {
   async entry (key) {
     const { keyname, filename } = keyResolve(this.root, key)
 
-    const stat = await lstat(filename)
-    if (!stat || stat.isDirectory()) {
+    const st = await (this.followLinks ? stat : lstat)(filename)
```
Comment (mafintosh) [inline index.js:32]: set stat once in the constructor

```
this._stat = opts.followLinks ? stat : lstat
```

Comment (LuKks) [reply index.js:32]: Awesome

**Pull request merged**